Functions to calculate themal radiation flux

MasanoriKanamaru opened the pull request:
# Abstract
This PR adds `src/radiation.jl` file to calculate thermal radiation from the simulated temperature distribution. This function does not affect the existing calculations.

# Updated test files
- `test/shape/fractal_v2572_f5000.obj`: This file is a shape model of a random fractal terrain used in the test at `test/blackbody_radiation.jl`.
<img width="918" alt="fractal_v2572_f5000" src="https://github.com/user-attachments/assets/c76f260c-f3a0-4c3f-802b-d17d74c57b7d" />

- `test/blackbody_radiation.jl`: This file includes tests to check if the new functions don't throw errors and if the output values are valid.

# Notes
Currently, some tests are incomplete. For example, we have only one shape for the tests. It would be ideal to add more shapes, but we will skip adding them for now.

MasanoriKanamaru committed:
Add `radiation.jl`

MasanoriKanamaru committed:
Add test for `radiation.jl`

MasanoriKanamaru committed:
Add a fractal topography model for test

MasanoriKanamaru committed:
Update radiation.jl

MasanoriKanamaru committed:
Update radiation.jl

MasanoriKanamaru committed:
Update radiation.jl

MasanoriKanamaru committed:
Update blackbody_radiation.jl

Add test for local TPM and thermal radiation calculation

hyrodium commented on src/radiation.jl:
```diff
@@ -0,0 +1,72 @@
+# Return
+- `radiance` : Spectral radiance [W/m²/m/steradian]
+
+c.f. https://github.com/JuliaAstro/Planck.jl/blob/main/src/Planck.jl
```
```suggestion
cf. https://github.com/JuliaAstro/Planck.jl/blob/main/src/Planck.jl
```
https://en.wikipedia.org/wiki/Cf.

hyrodium commented on test/blackbody_radiation.jl:
```diff
@@ -0,0 +1,93 @@
+    """
+    println(msg)
+
+    @test AsteroidThermoPhysicalModels.blackbody_radiation(6e-7, 5850) ≈ 2.583616647617974e13
```
Could you add comments to describe the meanings of the values?

MasanoriKanamaru committed:
Update src/radiation.jl

Co-authored-by: Yuto Horikawa <hyrodium@gmail.com>

MasanoriKanamaru committed:
Update blackbody_radiation.jl

Add a comment

hyrodium commented on src/radiation.jl:
```diff
@@ -0,0 +1,72 @@
+- `temperatures` : Temperature of each facet of the shape model [K]
+- `obs`          : Position vector of the observer in the same coordinate system as `shape` [m]
+"""
+function thermal_radiation(shape, emissivities, temperatures, obs)
```
```suggestion
function thermal_radiation(shape::ShapeModel, emissivities::AbstractVector{<:Real}, temperatures::AbstractVector{<:Real}, obs::StaticVector{3, <:Real})
```
I think adding argument types would be helpful.

MasanoriKanamaru committed:
Update src/radiation.jl

Co-Authored-By: Yuto Horikawa <hyrodium@gmail.com>

MasanoriKanamaru committed:
Update blackbody_radiation.jl

Just add a comment.

MasanoriKanamaru committed:
Update radiation.jl

`L` denotes radiation [W/m²].

MasanoriKanamaru committed:
Update radiation.jl

Just correct some comments.

MasanoriKanamaru committed:
Update blackbody_radiation.jl

Added the value tests of the thermal radiation.

MasanoriKanamaru committed:
Rename `test/blackbody_radiation.jl` to `test/thermal_radiation.jl`

MasanoriKanamaru committed:
Change function names

- blackbody_radiation -> blackbody_radiance
- thermal_radiation -> thermal_radiance

MasanoriKanamaru committed:
Change file names

src/radiation.jl -> src/thermal_radiation.jl

MasanoriKanamaru committed:
Update thermal_radiation.jl

Add a test of blackbody radiation

MasanoriKanamaru committed:
Update AsteroidThermoPhysicalModels.jl

MasanoriKanamaru committed:
Update thermal_radiation.jl

hyrodium reviewed (approved):
LGTM!

MasanoriKanamaru committed:
Update thermal_radiation.jl
